src/nccl_ofi_rdma.c

```diff
@@ -5204,6 +5204,15 @@ static int connect(nccl_net_ofi_ep_t *base_ep,
 
 static void ep_rail_release(nccl_net_ofi_ep_rail_t *rail, int dev_id)
 {
+	if (ofi_nccl_endpoint_per_communicator() != 0) {
+		/* when using an endpoint per communicator with a shared cq
+		(instead of a cq per endpoint), set the rail->cq pointer to NULL
+		here so	that the cq isn't actually released in ep_release().
+		The cq will be released when the domain is cleaned up
+		(which it currently never is, because there's no
+		plugin fini() function */
```

Comment:
Well now there is. ([d4a0fea](https://github.com/aws/aws-ofi-nccl/commit/d4a0feadc3f3760c4da45e441b0db66db62df995)) 😄 Use this.